api/osf_groups/serializers.py

```diff
@@ -0,0 +1,191 @@
+from rest_framework import serializers as ser, exceptions
+
+from framework.auth.core import Auth
+from api.base.serializers import (
+    IDField,
+    LinksField,
+    JSONAPISerializer,
+    RelationshipField,
+    TypeField,
+    VersionedDateTimeField,
+)
+from api.base.utils import absolute_reverse
+from api.nodes.serializers import CompoundIDField
+from osf.models import OSFUser
+from osf.models.osf_group import OSFGroup
+from osf.utils.permissions import GROUP_ROLES, MEMBER, MANAGER
+
+
+class GroupSerializer(JSONAPISerializer):
+    filterable_fields = frozenset([
+        'name',
+    ])
+
+    id = IDField(source='_id', read_only=True)
+    type = TypeField()
+    name = ser.CharField(required=True)
+    date_created = VersionedDateTimeField(source='created', read_only=True)
+    date_modified = VersionedDateTimeField(source='modified', read_only=True)
+
+    links = LinksField({
+        'self': 'get_absolute_url',
+    })
+
+    def get_absolute_url(self, obj):
+        return obj.get_absolute_url()
+
+    members = RelationshipField(
+        related_view='groups:group-members',
+        related_view_kwargs={'group_id': '<_id>'},
+    )
+
+    class Meta:
+        type_ = 'groups'
+
+    def create(self, validated_data):
+        group = OSFGroup(creator=validated_data['creator'], name=validated_data['name'])
+        group.save()
+        return group
+
+    def update(self, instance, validated_data):
+        if 'name' in validated_data:
+            instance.set_group_name(validated_data.get('name'))
```

Comment:
You'll get a "This field can't be None, or similar" `name` is a CharField with required=True, so DRF will enforce that it can't be None or an empty string. That happens before this update method is called.